packages/core/src/routes/sso-connector/utils.ts

```diff
@@ -57,14 +65,26 @@ export const parseConnectorConfig = (
   Return undefined if failed to fetch or parse the config.
 */
 export const fetchConnectorProviderDetails = async (
-  connector: SupportedSsoConnector
+  connector: SupportedSsoConnector,
+  tenantId: string
 ): Promise<SsoConnectorWithProviderConfig> => {
-  const { providerName } = connector;
+  const { providerName, config } = connector;
 
   const { logo, constructor } = ssoConnectorFactories[providerName];
 
   const providerConfig = await trySafe(async () => {
-    const instance = new constructor(connector);
+    const instance = new constructor(connector, tenantId);
+
+    // To avoid `getConfig()` being called in integration tests and throwing time out error.
+    if (isIntegrationTest) {
```

Comment:
Could remove the mock logic here. 
Use local logto as OIDC issuer, and a valid SAML metadata file for testing.

Reply:
Updated, use XML-format SAML metadata for testing the parser.